components/brave_rewards/browser/rewards_notification_service_impl.cc

```diff
@@ -186,7 +187,8 @@ void RewardsNotificationServiceImpl::ReadRewardsNotifications(
     if (notification_id_opt)
       notification_id = *notification_id_opt;
     int notification_type = it->FindIntKey("type").value_or(0);
-    int notification_timestamp = it->FindIntKey("timestamp").value_or(0);
+    int notification_timestamp =
```

Comment:
Will this work for the upgrade case, when "timestamp" will be an int value? It seems like `base::ValueToInt64` will return `nullopt` and we'll get a zero.

One option might be to parse and store as a double instead, and then do the `unit64_t` <-> `double` conversions for the notification structure.

Reply:
Another option would be to use `base::ValueToTime`, which would store the timestamp in a human readable format.

Reply:
yes, I think this will cause problems on "upgrade". Probably best to cast to int for now like we used to do implicitly via SetInteger in StoreRewardsNotifications and leave this line as it was. Then this can be addressed with a follow up issue as you suggested in https://github.com/brave/brave-core/pull/13498#discussion_r918379403

Reply:
Created https://github.com/brave/brave-browser/issues/24015 and reverted this to explicit conversion to int in 1b65ee45ed508f90ffc0bf14de436e483ab88e3f